File: javaclient/app/src/main/java/javaclient/app/models/MetricsThisWeek.java
package javaclient.client.models;

import com.microsoft.kiota.serialization.AdditionalDataHolder;
import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
/**
 * Summary totals for the last 7 days
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class MetricsThisWeek implements AdditionalDataHolder, Parsable {
    /**
     * APIs added in the last week
     */
    private Integer added;
    /**
     * Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     */
    private Map<String, Object> additionalData;
    /**
     * APIs updated in the last week
     */
    private Integer updated;
    /**
     * Instantiates a new {@link MetricsThisWeek} and sets the default values.
     */
    public MetricsThisWeek() {
        this.setAdditionalData(new HashMap<>());
    }
    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param parseNode The parse node to use to read the discriminator value and create the object
     * @return a {@link MetricsThisWeek}
     */
    @jakarta.annotation.Nonnull
    public static MetricsThisWeek createFromDiscriminatorValue(@jakarta.annotation.Nonnull final ParseNode parseNode) {
        Objects.requireNonNull(parseNode);
        return new MetricsThisWeek();
    }
    /**
     * Gets the added property value. APIs added in the last week
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getAdded() {
        return this.added;
    }
    /**
     * Gets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @return a {@link Map<String, Object>}
     */
    @jakarta.annotation.Nonnull
    public Map<String, Object> getAdditionalData() {
        return this.additionalData;
    }
    /**
     * The deserialization information for the current model
     * @return a {@link Map<String, java.util.function.Consumer<ParseNode>>}
     */
    @jakarta.annotation.Nonnull
    public Map<String, java.util.function.Consumer<ParseNode>> getFieldDeserializers() {
        final HashMap<String, java.util.function.Consumer<ParseNode>> deserializerMap = new HashMap<String, java.util.function.Consumer<ParseNode>>(2);
        deserializerMap.put("added", (n) -> { this.setAdded(n.getIntegerValue()); });
        deserializerMap.put("updated", (n) -> { this.setUpdated(n.getIntegerValue()); });
        return deserializerMap;
    }
    /**
     * Gets the updated property value. APIs updated in the last week
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getUpdated() {
        return this.updated;
    }
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public void serialize(@jakarta.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeIntegerValue("added", this.getAdded());
        writer.writeIntegerValue("updated", this.getUpdated());
        writer.writeAdditionalData(this.getAdditionalData());
    }
    /**
     * Sets the added property value. APIs added in the last week
     * @param value Value to set for the added property.
     */
    public void setAdded(@jakarta.annotation.Nullable final Integer value) {
        this.added = value;
    }
    /**
     * Sets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public void setAdditionalData(@jakarta.annotation.Nullable final Map<String, Object> value) {
        this.additionalData = value;
    }
    /**
     * Sets the updated property value. APIs updated in the last week
     * @param value Value to set for the updated property.
     */
    public void setUpdated(@jakarta.annotation.Nullable final Integer value) {
        this.updated = value;
    }
}

File: goclient/goclient/models/metrics_this_week.go
package models

import (
    i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91 "github.com/microsoft/kiota-abstractions-go/serialization"
)

// Metrics_thisWeek summary totals for the last 7 days
type Metrics_thisWeek struct {
    // APIs added in the last week
    added *int32
    // Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    additionalData map[string]any
    // APIs updated in the last week
    updated *int32
}
// NewMetrics_thisWeek instantiates a new Metrics_thisWeek and sets the default values.
func NewMetrics_thisWeek()(*Metrics_thisWeek) {
    m := &Metrics_thisWeek{
    }
    m.SetAdditionalData(make(map[string]any))
    return m
}
// CreateMetrics_thisWeekFromDiscriminatorValue creates a new instance of the appropriate class based on discriminator value
// returns a Parsable when successful
func CreateMetrics_thisWeekFromDiscriminatorValue(parseNode i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable, error) {
    return NewMetrics_thisWeek(), nil
}
// GetAdded gets the added property value. APIs added in the last week
// returns a *int32 when successful
func (m *Metrics_thisWeek) GetAdded()(*int32) {
    return m.added
}
// GetAdditionalData gets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
// returns a map[string]any when successful
func (m *Metrics_thisWeek) GetAdditionalData()(map[string]any) {
    return m.additionalData
}
// GetFieldDeserializers the deserialization information for the current model
// returns a map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error) when successful
func (m *Metrics_thisWeek) GetFieldDeserializers()(map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error)) {
    res := make(map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error))
    res["added"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetAdded(val)
        }
        return nil
    }
    res["updated"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetUpdated(val)
        }
        return nil
    }
    return res
}
// GetUpdated gets the updated property value. APIs updated in the last week
// returns a *int32 when successful
func (m *Metrics_thisWeek) GetUpdated()(*int32) {
    return m.updated
}
// Serialize serializes information the current object
func (m *Metrics_thisWeek) Serialize(writer i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.SerializationWriter)(error) {
    {
        err := writer.WriteInt32Value("added", m.GetAdded())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("updated", m.GetUpdated())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteAdditionalData(m.GetAdditionalData())
        if err != nil {
            return err
        }
    }
    return nil
}
// SetAdded sets the added property value. APIs added in the last week
func (m *Metrics_thisWeek) SetAdded(value *int32)() {
    m.added = value
}
// SetAdditionalData sets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
func (m *Metrics_thisWeek) SetAdditionalData(value map[string]any)() {
    m.additionalData = value
}
// SetUpdated sets the updated property value. APIs updated in the last week
func (m *Metrics_thisWeek) SetUpdated(value *int32)() {
    m.updated = value
}
type Metrics_thisWeekable interface {
    i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.AdditionalDataHolder
    i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable
    GetAdded()(*int32)
    GetUpdated()(*int32)
    SetAdded(value *int32)()
    SetUpdated(value *int32)()
}
